spec/lib/tasks/kids_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "kids:promote_minor_accounts", type: :task do
+  let(:organization) { create(:organization, available_authorizations: %w(dummy_authorization_handler)) }
+  let(:user) { create(:user, :admin, :confirmed, organization:) }
+  let(:other_user) { create(:user, :admin, :confirmed, organization:) }
+  let(:minor) { create(:minor, tutor: user, organization:) }
+  let(:minor_to_promote) { create(:minor_to_promote, tutor: user, organization:) }
```

Comment:
```suggestion
  let(:minor_to_promote) { create(:minor, tutor: user, minor_data_to_promote:, organization:) }
  let(:minor_data_to_promote) { create :minor_data, birthday: }
  let(:birthday) { 14.years.ago }
```
I found this extra factory confusing, can we just set the birthday instead?
This also would let you check several ages. 
At least we also should test the consecutive age when an account is promoted (from 13 to 14 yo)

Reply:
I've tried to replicate the suggestions. In this specific case, I wasn't able to create "minor_to_promote" by creating another "minor" passing a "birthday" in its "minor_data_to_promote". "minor_data_to_promote" creates a ":minor_data" and this one a new minor user who creates another ":minor_data" and overrides the given "birthday"

Reply:
Right, the problem is that the factory `minor` is just a convenient way to create a "minor user". In your case,you just need to define all the properties of what makes a user a "minor":

```ruby
  let!(:minor_to_promote) { create(:user, :confirmed, organization:) }
  let!(:minor_to_promote_account) { create :minor_account, tutor: user, minor: minor_to_promote}
  let!(:minor_data_to_promote) { create :minor_data, user: minor_to_promote, birthday: }
  let(:birthday) { 15.years.ago }
```

Btw, once you do this, you can add a context and test the case for 14 years old.

Reply:
Done